src/test/groovy/pl/tomaszdziurko/jvm_bloggers/blog_posts/domain/BlogPostRepositorySpec.groovy

```diff
@@ -0,0 +1,65 @@
+package pl.tomaszdziurko.jvm_bloggers.blog_posts.domain
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.domain.PageRequest
+import org.springframework.test.context.ActiveProfiles
+import pl.tomaszdziurko.jvm_bloggers.SpringContextAwareSpecification
+import pl.tomaszdziurko.jvm_bloggers.blogs.domain.Blog
+import pl.tomaszdziurko.jvm_bloggers.blogs.domain.BlogRepository
+
+import java.time.LocalDateTime
+
+import static pl.tomaszdziurko.jvm_bloggers.blogs.domain.BlogType.PERSONAL
+
+@ActiveProfiles("test")
+class BlogPostRepositorySpec extends SpringContextAwareSpecification {
+
+    static NOT_MODERATED = null
+    static APPROVED = Boolean.TRUE
+    static REJECTED = Boolean.FALSE
+
+    @Autowired
+    BlogPostRepository blogPostRepository
+
+    @Autowired
+    BlogRepository blogRepository
+
+    def "Should order latest posts by moderation and publication date"() {
+        given:
+            def blog = aBlog()
+
+            def blogPosts = [
+                aBlogPost(1, LocalDateTime.of(2016, 1, 1, 12, 00), REJECTED, blog),
+                aBlogPost(2, LocalDateTime.of(2016, 1, 4, 12, 00), REJECTED, blog),
+                aBlogPost(3, LocalDateTime.of(2016, 1, 2, 12, 00), APPROVED, blog),
+                aBlogPost(4, LocalDateTime.of(2016, 1, 5, 12, 00), APPROVED, blog),
+                aBlogPost(5, LocalDateTime.of(2016, 1, 3, 12, 00), NOT_MODERATED, blog),
+                aBlogPost(6, LocalDateTime.of(2016, 1, 6, 12, 00), NOT_MODERATED, blog)
+            ]
+
+            blogPosts.each { blogPost -> blogPostRepository.save(blogPost) }
```

Comment:
I guess you can use `blogPostRepository.save(blogPosts)` instead. See `[JpaRepository.save(Iterable)]`(http://docs.spring.io/spring-data/jpa/docs/current/api/org/springframework/data/jpa/repository/JpaRepository.html#save-java.lang.Iterable-) for reference.